src/main/webapp/data-visualization.js

```diff
@@ -25,8 +25,8 @@ async function getGeoData(location, isCountyQuery) {
 async function displayVisualization(censusDataArray, description,
   location, isCountyQuery) {
   document.getElementById('colors').style.display = 'block';
-  setColor();
-  const color = localStorage.getItem('color');
+  const color = setColor();
+  document.getElementById('setColor').value = color;
```

Comment:
Since you now moved the 'set' part here the function should actually be called getColor now.

Reply:
true! fixed.